tuning/data/data_processors.py

```diff
@@ -0,0 +1,238 @@
+
+                    column_names = raw_datasets[splitName].column_names
+
+                    # remove __content__ from all processing
+                    if "__content__" in column_names:
+                        column_names.remove("__content__")
+
+                    if "remove_columns" not in kwargs:
+                        kwargs["remove_columns"] = None
+                    if kwargs["remove_columns"] == "all":
+                        kwargs["remove_columns"] = column_names
+
+                    if "num_proc" not in kwargs:
+                        kwargs["num_proc"] = os.cpu_count()
+
+                    if "fn_kwargs" not in kwargs:
+                        kwargs["fn_kwargs"] = {}
+
+                    kwargs["fn_kwargs"]["tokenizer"] = self.tokenizer
+                    kwargs["fn_kwargs"]["column_names"] = column_names
+
+                    kwargs["fn_kwargs"] = dict(kwargs["fn_kwargs"], **extra_kwargs)
+
+                    logging.info("Applying Handler: %s Args: %s", data_handler, kwargs)
+
+                    raw_datasets = raw_datasets.map(handler, **kwargs)
+
+            if final_datasets is None:
+                final_datasets = raw_datasets
+            else:
+                for k in raw_datasets.keys():
+                    if k in final_datasets:
+                        final_datasets[k] = datasets.concatenate_datasets(
+                            [final_datasets[k], raw_datasets[k]]
+                        )
+                    else:
+                        final_datasets[k] = raw_datasets[k]
+
+        if "train" in final_datasets:
+            train_dataset = final_datasets["train"]
+
+        return train_dataset
+
+    def process_dataset_configs(
+        self, dataset_configs: List[DataSetConfig], **kwargs
+    ) -> Union[Dataset, IterableDataset]:
+        train_dataset = None
+
+        if torch.distributed.is_available() and torch.distributed.is_initialized():
+            if torch.distributed.get_rank() == 0:
+                logging.info("Processing data on rank 0...")
+                train_dataset = self._process_dataset_configs(dataset_configs, **kwargs)
+            else:
+                train_dataset = None
+
+            # Use broadcast_object_list to share the dataset object across ranks
+            # TODO: Check if torch.distributed.barrier() is called in broadcast_object_list()
+            obj_list = [train_dataset]
+            torch.distributed.broadcast_object_list(obj_list, src=0)
+            train_dataset = obj_list[0]
+        else:
+            logging.info("Processing data...")
+            train_dataset = self._process_dataset_configs(dataset_configs, **kwargs)
+
+        return train_dataset
+
+
+def autoregister_available_handlers(processor: DataPreProcessor):
+    if processor is None:
+        return
+    for name, func in AVAILABLE_DATA_HANDLERS.items():
+        processor.register_data_handler(name=name, func=func)
+
+
+def get_dataprocessor(
```

Comment:
the function says `dataprocessro` but it returns a `dataPreProcessor` why the difference?